src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/EditorConfigFileGenerator_NamingStyles.cs

```diff
@@ -10,23 +10,60 @@
 using Microsoft.CodeAnalysis.CodeStyle;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+using Microsoft.CodeAnalysis.NamingStyles;
 
 namespace Microsoft.CodeAnalysis.Options
 {
     internal static partial class EditorConfigFileGenerator
     {
+        public static void AppendNamingStylePreferencesToEditorConfig(IEnumerable<NamingRule> namingRules, StringBuilder editorconfig, string? language = null)
+        {
+            var symbolSpecifications = namingRules.Select(x => x.SymbolSpecification).ToImmutableArray();
+            var namingStyles = namingRules.Select(x => x.NamingStyle).ToImmutableArray();
```

Comment:
It's probably not a huge deal, but doing `ToImmutableArray()` here is slightly more expensive than just leaving as an `IEnumerable` since you're going to just enumerate again and not use the `ImmutableArray`. That ore combine with the `Select` below for one chained `LINQ`